.travis.yml

```diff
@@ -1,122 +1,117 @@
+# We need to use TEST=none to remove the test-data submodule.
+# Barebone build with no test data
+env: INSTALL_TYPE=develop TEST=none NUMPYVER=1.11 TRAVIS_PYTHON_VERSION="2.7"
+
 matrix:
-  allow_failures:
-      - python: "3.5"
   fast_finish: true
   include:
-    - env: XSPECVER="12.9.0i" FITS="astropy" INSTALL_TYPE=install TEST=smoke MATPLOTLIBVER=1.5
-      python: "2.7"
+    # Full build (including ds9 and xspec), Python 2.7, setup.py develop, astropy, matplotlib 1.5
+    - env: XSPECVER="12.9.0i" FITS="astropy" INSTALL_TYPE=develop TEST=submodule MATPLOTLIBVER=1.5 TRAVIS_PYTHON_VERSION="2.7"
       sudo: required
       dist: trusty
-    - env: XSPECVER="12.8.2q" FITS="astropy" INSTALL_TYPE=develop TEST=submodule MATPLOTLIBVER=1.5
-      python: "2.7"
+    # As above, Python 3.5
+    - env: XSPECVER="12.9.0i" FITS="astropy" INSTALL_TYPE=develop TEST=submodule MATPLOTLIBVER=1.5 TRAVIS_PYTHON_VERSION="3.5"
       sudo: required
       dist: trusty
-    - env: XSPECVER="12.9.0i" FITS="astropy" INSTALL_TYPE=develop TEST=submodule MATPLOTLIBVER=1.5 MKL=nomkl
-      python: "2.7"
+    # As above, Python 3.6, setup.py install
+    - env: XSPECVER="12.9.0i" FITS="astropy" INSTALL_TYPE=install TEST=submodule MATPLOTLIBVER=1.5 TRAVIS_PYTHON_VERSION="3.6"
       sudo: required
       dist: trusty
-    - env: XSPECVER="12.9.0i" FITS="astropy" INSTALL_TYPE=develop TEST=submodule MATPLOTLIBVER=1.5
-      python: "3.5"
-      sudo: required
-      dist: trusty
-    - env: XSPECVER="12.9.0i" FITS="astropy" INSTALL_TYPE=develop TEST=submodule MATPLOTLIBVER=1.5
-      python: "2.7"
-      sudo: required
-      dist: trusty
-    - env: INSTALL_TYPE=install TEST=package MATPLOTLIBVER=1.5
-      python: "2.7"
-    - env: FITS="astropy" INSTALL_TYPE=develop TEST=submodule MATPLOTLIBVER=1.5
-      python: "2.7"
-    - env: FITS="astropy" INSTALL_TYPE=install TEST=package MATPLOTLIBVER=1.4 NUMPYVER=1.9
-      python: "2.7"
-    - env: FITS="astropy" INSTALL_TYPE=install TEST=package MATPLOTLIBVER=1.5 NUMPYVER=1.10
-      python: "2.7"
-    - env: INSTALL_TYPE=install TEST=package NUMPYVER=1.11
-      python: "2.7"
+    # Install sherpatest package rather than relying on relative location of the submodule
+    # Also, install matplotlib 2.0 and do not install astropy (checks tests are properly skipped)
+    - env: INSTALL_TYPE=install TEST=package MATPLOTLIBVER=2.0 TRAVIS_PYTHON_VERSION="2.7"
+    # Install astropy without matplotlib (checks tests are properly skipped)
+    - env: INSTALL_TYPE=develop TEST=submodule FITS="astropy" TRAVIS_PYTHON_VERSION="3.5"
   allow_failures:
-    - env: XSPECVER="12.8.2q" FITS="astropy" INSTALL_TYPE=develop TEST=submodule MATPLOTLIBVER=1.5
-
-env: FITS="pyfits" INSTALL_TYPE=develop TEST=submodule MATPLOTLIBVER=1.4 # 1.5 does not support numpy 1.8, required by pyfits
+    - env: XSPECVER="12.9.0i" FITS="astropy" INSTALL_TYPE=install TEST=submodule MATPLOTLIBVER=1.5 TRAVIS_PYTHON_VERSION="3.6"
```

Comment:
Ahhh. I had taken "allow failures" to be a place that you can specify runs that can fail, rather than this subtly-different meaning.

We need a comment in the allow-failures section to say why we allow this to fail (so we remember in 6 months time when we are reviewing the build).